src/luarocks/cmd.lua

```diff
@@ -85,7 +85,7 @@ do
             replace_tree(args, root_dir)
          end
       elseif args["local"] then
-         if not cfg.home_tree then
+         if os.getenv("USER") == "root" then
```

Comment:
I agree the previous test was done in a roundabout way and being explicit about it is better, but this solution above is Unix-specific, so it should be abstracted away. Maybe a `fs.is_superuser` function? Or `cfg.is_superuser` in case this is too early and `fs` platforms are not configured yet.

I understand the condition would only hit in a Unix system, but for future maintenance is better to not add platform-dependent code on the platform-independent modules.

Reply:
Good suggestion! I'll try implementing such a function.